### PR TITLE
use HingeLoss as default loss function

### PR DESCRIPTION
**Pull request by feiyun0112**
#6815 

[StandardTrainersCatalog.AveragedPerceptron](https://github.com/dotnet/machinelearning/blob/09b80f8a08340dc7d79ac75e13c722313f0845eb/src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs#L429C13-L429C196) factory method uses LogLoss as its default loss function, which contradicts method documentation and defaults.AveragedPerceptronTrainer.Options

### CODE REVIEW

File: src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs [hunk 1]
```diff
@@ -184,7 +184,7 @@ internal AveragedPerceptronTrainer(IHostEnvironment env, Options options)
                 DecreaseLearningRate = decreaseLearningRate,
                 L2Regularization = l2Regularization,
                 NumberOfIterations = numberOfIterations,
-                LossFunction = lossFunction ?? new HingeLoss()
```
Comment (ericstj) [inline src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs:187]: Doesn't this remove the `HingeLoss` default?

Comment (feiyun0112) [reply src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs:187]: it will default by `LossFunctionFactory`

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L156C36-L156C36](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L156C36-L156C36)

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L119C71-L119C71](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L119C71-L119C71)

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L100](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L100)

**Comment by michaelgsharp** [comment]
@feiyun0112 why do you want to switch the default loss function? Is HingeLoss better?

**Comment by feiyun0112** [comment]
> @feiyun0112 why do you want to switch the default loss function? Is HingeLoss better?

please see this issue https://github.com/dotnet/machinelearning/issues/6815
